### ISSUE TITLE
Exception when using fakeredis on Py3

### ISSUE THREAD

**Issue by si14** [niwinz/django-redis#212]
Here is a full stacktrace:

```
[2016-08-22 16:37:39,139][ERROR] django.request 47831-123145312813056 base.py:handle_uncaught_exception:284 Internal Server Error: /en/
Traceback (most recent call last):
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/core/handlers/base.py", line 174, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/core/handlers/base.py", line 172, in get_response
    response = response.render()
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/response.py", line 160, in render
    self.content = self.rendered_content
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/response.py", line 137, in rendered_content
    content = template.render(context, self._request)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/backends/django.py", line 95, in render
    return self.template.render(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/base.py", line 206, in render
    return self._render(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/loader_tags.py", line 173, in render
    return compiled_parent._render(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/loader_tags.py", line 69, in render
    result = block.nodelist.render(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django/template/library.py", line 201, in render
    output = self.func(*resolved_args, **resolved_kwargs)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/cachalot/templatetags/cachalot.py", line 18, in get_last_invalidation
    return get_last_invalidation_function(*tables_or_models, **kwargs)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/cachalot/api.py", line 107, in get_last_invalidation
    cache_alias, db_alias).get_many(table_cache_keys).values()
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django_redis/cache.py", line 32, in _decorator
    return method(self, *args, **kwargs)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django_redis/cache.py", line 107, in get_many
    return self.client.get_many(*args, **kwargs)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/django_redis/client/default.py", line 337, in get_many
    results = client.mget(*new_keys)
  File "/Users/si14/.virtualenvs/melsciwebpy3/lib/python3.5/site-packages/fakeredis.py", line 381, in mget
    for key in all_keys:
TypeError: 'CacheKey' object is not iterable
```

After some poking around, I think the reason is that `django-redis` uses `CacheKey` as a key (https://github.com/niwinz/django-redis/blob/master/django_redis/util.py#L33), while `fakeredis` has a lot of checks like `isinstance(keys, string_types)` (e.g.  https://github.com/jamesls/fakeredis/blob/master/fakeredis.py#L1454-L1463) `CacheKey` isn't a subclass of `str`, so `fakeredis` breaks.

**Comment by niwinz** [comment]
Hmm, but redis-py works correctly with `CacheKey` so I think it should be solved on fakeredis in order to be fully compliant with `redis-py` library.

In any case, making `CacheKey` as subclass of str I think it is not goind to be a big problem, so we can facilitate it at django-redis side.

**Comment by si14** [comment]
Proposed fix: #216
